Fix test pull verified

dmcgowan opened the pull request:
Update pull code to consider any layer download or new tag as an update.
Update hello-world frozen image to be explicitly tagged as frozen, to not interfere with pull tests.  The hello-world is used by pull tests because of its small size and there is no other official image with such a size.

fixes #11383

dmcgowan committed:
Fix test pull verified

Update pull code to consider any layer download or new tag as an update.
Update hello-world frozen image to be explicitly tagged as frozen, to not interfere with pull tests.  The hello-world is used by pull tests because of its small size and there is no other official image with such a size.

fixes #11383

Signed-off-by: Derek McGowan <derek@mcgstyle.net> (github: dmcgowan)

tianon commented on hack/make/.ensure-frozen-images:
```diff
@@ -4,7 +4,7 @@ set -e
 images=(
 	busybox:latest
-	hello-world:latest
+	hello-world:frozen
```
I love the idea, but this won't actually work because the last block of the `if` in this file is the "if all else fails, `docker pull <image>`", which will fail to pull the `frozen` tag here. :disappointed:

dmcgowan replied:
All else fails meaning that pull directly by image ID failed? If the image is intended to be pinned at a particular version, I don't see why we would want to `docker pull <image>` or `docker pull <image>:<tag>`.  Given the current circumstances though I will explore another way to fix this part of the issue.

tianon replied:
All else failing meaning that we don't have the pre-downloaded images from the `Dockerfile` (which is the ideal case) _and_ we don't have `curl` in order to use the download script directly (which is the second-best option).  Our last-resort fallback in this file is to `docker pull` (see line 39 of this file).

I'd be OK making that a hard fail instead, but I think that would warrant a little bit more discussion, because `curl` would essentially become a build-time dep for anyone wanting to run the integration-cli tests.

tianon committed:
Enforce our use of the explicitly frozen images

This requires that any environment where we wish to run the integration-cli tests includes both the `Dockerfile` and `curl`, which has been deemed an appropriate and acceptable trade-off.

Signed-off-by: Andrew "Tianon" Page <admwiggin@gmail.com>

tianon commented:
https://github.com/dmcgowan/docker/pull/20 :heart:

tianon commented:
(discussed with @jfrazelle and she agrees that this is fine)

dmcgowan commented:
Thanks for PR, included the commit

jfrazelle commented:
cool so if its fix this should be green :D

tianon commented:
LGTM, but I feel like mine is implied now :+1:

jfrazelle commented:
ugh ```
--- FAIL: TestEventsImagePull (0.84s)
	docker_cli_events_test.go:157: pulling the hello-world image from has failed: Pulling repository hello-world
		time="2015-03-16T21:13:06Z" level=fatal msg="Repository not found" 
		, exit status 1
``` 
is something else right

jfrazelle commented:
I think it is they are dubugging on registry side now, LGTM
